rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll

```diff
@@ -0,0 +1,47 @@
+private import rust
+import ControlFlowGraphImplSpecific::CfgImpl
+import Completion
+
+class CallTree extends StandardPostOrderTree instanceof Call {
+  override ControlFlowTree getChildNode(int i) { result = super.getArg(i) }
+}
+
+class BinaryOpTree extends StandardPostOrderTree instanceof BinaryOp {
+  override ControlFlowTree getChildNode(int i) {
+    i = 0 and result = super.getLhs()
+    or
+    i = 1 and result = super.getRhs()
+  }
+}
+
+class IfTree extends PostOrderTree instanceof If {
+  override predicate first(AstNode node) { first(super.getCondition(), node) }
+
+  override predicate propagatesAbnormal(AstNode child) { none() }
```

Comment:
I think this should be `child = [super.getCondition(), super.getThen(), super.getElse()]`; for example `return`s are also considered "abnormal".